src/mgmt/srv_query.c

```diff
@@ -364,6 +364,32 @@ add_ctrlr_details(Ctl__NvmeController *ctrlr, struct bio_dev_info *dev_info)
 		ctrlr->model, ctrlr->serial, ctrlr->fw_rev, ctrlr->vendor_id, ctrlr->pci_dev_type,
 		ctrlr->socket_id);
 
+	/* Populate NVMe namespace id and capacity */
+
+	if (dev_info->bdi_ctrlr->nss == NULL) {
+		D_ERROR("nss not initialized in bio_dev_info");
+		return -DER_INVAL;
+	}
+	D_ASSERT(dev_info->bdi_ctrlr->nss->next == NULL);
+
+	/* When describing a SMD, only one NVMe namespace is relevant */
+	D_ALLOC_ARRAY(ctrlr->namespaces, 1);
+	if (ctrlr->namespaces == NULL) {
+		return -DER_NOMEM;
+	}
+	D_ALLOC_PTR(ctrlr->namespaces[0]);
+	if (ctrlr->namespaces[0] == NULL) {
+		return -DER_NOMEM;
```

Comment:
[defect] ctrlr->namespaces leaked?

Reply:
yes, thanks for spotting, fixed in the relevant free function